AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -553,8 +551,13 @@ - (void)_loadViewOrLayerIsLayerBacked:(BOOL)isLayerBacked
   if (isLayerBacked) {
     TIME_SCOPED(_debugTimeToCreateView);
     _layer = [self _layerToLoad];
-    // Surpress warning for Base SDK > 10.0
-    _layer.delegate = (id<CALayerDelegate>)self;
+    static int ASLayerDelegateAssociationKey;
+
+    // We cannot make self the layer's delegate directly, because the delegate is
```

Comment:
Maybe say "CALayer's .delegate property is documented to be weak, but the code emitted is actually assign."
"Because ASDisplayNodes may begin deallocation on a background thread but only execute -dealloc on the main thread, the only way to avoid a dangling pointer is to use a weak proxy as the CALayer .delegate"